tests/network/test_payload_checks.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+import phantom as ph
+
+
+class AgentA(ph.Agent):
+    pass
+
+
+class AgentB(ph.Agent):
+    pass
+
+
+@ph.msg_payload()
+class Payload1:
+    pass
+
+
+@ph.msg_payload(AgentA, AgentB)
+class Payload2:
+    pass
+
+
+def test_payload_checks():
```

Comment:
can you add a test for: 
- list of classes
- sender type and no receiver type
- receiver type and no sender type